ocean_provider/file_types/file_types.py

```diff
@@ -62,3 +66,41 @@ def get_download_url(self):
             raise Exception("No IPFS_GATEWAY defined, can not resolve ipfs hash.")
 
         return urljoin(os.getenv("IPFS_GATEWAY"), urljoin("ipfs/", self.hash))
+
+    @enforce_types
+    def get_filename(self):
+        return uuid4().hex
+
+
+class ArweaveFile(EndUrlType, FilesType):
+    @enforce_types
+    def __init__(
+        self,
+        transactionId: Optional[str] = None,
+        headers: Optional[dict] = None,
+        userdata=None,
+    ) -> None:
+        self.transactionId = transactionId
+        self.type = "arweave"
+        self.headers = headers if headers else {}
+        self.userdata = userdata
+        self.method = "get"
+
+    @enforce_types
+    def validate_dict(self) -> Tuple[bool, Any]:
+        if not self.transactionId:
+            return False, "malformed service files, missing required keys."
```

Comment:
I think you can be more specific here, and say that the transactionId is missing :)

Reply:
Done.